app/src/main/java/io/github/danthe1st/jdoc4droid/activities/list/classes/ListClassesActivity.java

```diff
@@ -72,6 +80,23 @@ public void onCreate(Bundle savedInstanceState) {
         });
     }
 
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        boolean ret=super.onCreateOptionsMenu(menu);
+        filterButton=menu.findItem(R.id.app_bar_filter);
+        filterButton.setVisible(true);
+        return ret;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
```

Comment:
This methods is also called by other elements of the options menu.
Please run that code only when the selected item is the filer:
```java
    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if(item.getItemId()==R.id.app_bar_filter){
            final String title = String.valueOf(item.getTitle());
            if (getAvailableFilters().contains(title)){
                onFilterChange(title);
            }
            return true;
        }else{
            return super.onOptionsItemSelected(item);
        }
    }
```